tasks/get_current_archive_list/get_current_archive_list.py

```diff
@@ -519,39 +519,39 @@ def handler(event: Dict[str, List], context) -> Dict[str, Any]:
     Environment Vars:
         INTERNAL_REPORT_QUEUE_URL (string): The URL of the SQS queue the job came from.
         S3_CREDENTIALS_SECRET_ARN (string): The ARN of the secret containing s3 credentials.
+        DB_CONNECT_INFO_SECRET_ARN (string): Secret ARN of the AWS secretsmanager secret for connecting to the database.
         See shared_db.py's get_configuration for further details.
 
     Returns: See output.json for details.
     """
     LOGGER.setMetadata(event, context)
 
-    try:
-        internal_report_queue_url = str(
-            os.environ[OS_ENVIRON_INTERNAL_REPORT_QUEUE_URL_KEY]
-        )
-    except KeyError as key_error:
-        LOGGER.error(
-            f"{OS_ENVIRON_INTERNAL_REPORT_QUEUE_URL_KEY} environment value not found."
-        )
-        raise key_error
-
-    try:
-        s3_credentials_secret_arn = str(
-            os.environ[OS_ENVIRON_S3_CREDENTIALS_SECRET_ARN_KEY]
-        )
-    except KeyError as key_error:
-        LOGGER.error(
-            f"{OS_ENVIRON_S3_CREDENTIALS_SECRET_ARN_KEY} environment value not found."
-        )
-        raise key_error
+    env_variable_list=[
```

Comment:
This implementation is not the best way to approach this. So a couple of things to think about here, first, you are using array position to determine value which obfuscates what you are passing. Second, your initial env_variable_list data structure is a bit confusing and does not really add any value to the code below. A better data structure maybe a dictionary or collection. In addition, you are doing a `str()` function call on `os.environ()` call which already returns a string. If you wanted to stay with a looping structure, one way you may approach this maybe something like the following:

```
env_dict = {
    OS_ENVIRON_INTERNAL_REPORT_QUEUE_URL_KEY: None,
    OS_ENVIRON_S3_CREDENTIALS_SECRET_ARN_KEY: None,
    OS_ENVIRON_DB_CONNECT_INFO_SECRET_ARN_KEY: None
}

for key in env_dict:
    try:
        env_dict[key] = os.environ[key]
        # This is a very basic check to make sure an empty string is not set. A better check would employ
        # a regex expression to make sure it is not full of spaces
        if len(env_dict[key]) == 0 or env_dict[key] is None:
            raise KeyError(f"Empty value for {key}")
    except KeyError as key_error:
        LOGGER.error(f"{key} environment value not found.")
        raise
        
# Calls then would be something like
db_connect_info = shared_db.get_configuration(env_dict[OS_ENVIRON_DB_CONNECT_INFO_SECRET_ARN_KEY])
```

A potentially better approach may be to extract the logic into a function like the following:

```
def check_env_variable(env_name: str)->str:
    try:
        env_value = os.environ[env_name]
        # This is a very basic check to make sure an empty string is not set. A better check would employ
        # a regex expression to make sure it is not full of spaces
        if len(env_value) == 0 or env_value is None:
            raise KeyError(f"Empty value for {env_name}")
    except KeyError as key_error:
        LOGGER.error(f"{env_name} environment value not found.")
        raise
        
# Calls then would be something like
db_connect_info = shared_db.get_configuration(check_env_variable(OS_ENVIRON_DB_CONNECT_INFO_SECRET_ARN_KEY))
```

Either approach works and there are additional ways to make this cleaner and more maintainable for the future which come down to the data structure you choose and other items.

Reply:
updated the code, unit tests and API.md file using the latter solution